packages/client/src/client.ts

```diff
@@ -0,0 +1,165 @@
+import { jsonOptions, PenumbraService } from '@penumbra-zone/protobuf';
+import { createPromiseClient, PromiseClient } from '@connectrpc/connect';
+
+import { assertPenumbra, assertProviderManifest, assertProviderRecord } from './assert.js';
+import { PenumbraProviderNotConnectedError } from './error.js';
+import { PenumbraStateEventDetail, PenumbraStateEvent, isPenumbraStateEvent } from './event.js';
+import { availableOrigin } from './get.js';
+import { PenumbraProvider } from './provider.js';
+import { PenumbraState } from './state.js';
+import {
+  ChannelTransportOptions,
+  createChannelTransport,
+} from '@penumbra-zone/transport-dom/create';
+
+export interface IPenumbraClient {
+  /**
+   * Asks users to approve the connection to a specific browser manifest URL.
+   * If `manifest` argument is not provided, tries to connect to the first injected provider.
+   * Returns the manifest url of the connected provider, or an error
+   */
+  readonly connect: (providerUrl?: string) => Promise<string>;
+
+  /**
+   * Tries to reconnect to the injected provider without asking for user approval.
+   */
+  readonly reconnect: (providerUrl?: string) => Promise<string>;
+
+  /** Reexports the `disconnect` function from injected provider */
+  readonly disconnect: () => Promise<void>;
+  /** Reexports the `isConnected` function from injected provider  */
+  readonly isConnected: () => boolean | undefined;
+  /** Reexports the `state` function from injected provider */
+  readonly getState: () => PenumbraState;
+  /** Provides a simplified callback interface to `PenumbraStateEvent`s. */
+  readonly onConnectionChange: (cb: (connection: PenumbraStateEventDetail) => void) => void;
+
+  /** Returns a new or re-used `PromiseClient<T>` for a specific `PenumbraService` */
+  readonly service: <T extends PenumbraService>(
+    service: T,
+    options: Omit<ChannelTransportOptions, 'getPort'>,
```

Comment:
transport options parameter should be optional - it's only necessary to configure a custom timeout or custom ser/de behavior. a good set of defaults is available in #1648